### ISSUE TITLE
SCRAM support

### ISSUE THREAD

**Issue by keithf4** [wrouesnel/postgres_exporter#231]
Appears that postgres_exporter doesn't support when SCRAM password authentication is set in pg_hba.conf. 

https://www.postgresql.org/docs/10/auth-methods.html#AUTH-PASSWORD

```
Nov  8 10:52:30 localhost postgres_exporter: time="2018-11-08T10:52:30-05:00" level=info msg="Error opening connection to database (postgresql://ccp_monitoring:PASSWORD_REMOVED@localhost:5432/postgres?sslmode=disable): pq: unknown authentication response: 10" source="postgres_exporter.go:1030"
```
Thankfully just setting an override line above other entries in pg_hba.conf for the postgres_exporter role to md5 allows it to work for now even when the internal password storage is scram.

**Comment by ptman** [comment]
The support is needed in lib/pq: https://github.com/lib/pq/pull/608 or https://github.com/lib/pq/pull/788

**Comment by onderaycicek** [comment]
This issue solved. https://github.com/lib/pq/pull/833

**Comment by Neustradamus** [comment]
Linked to:
- https://github.com/lib/pq/issues/817
- https://github.com/lib/pq/pull/608
- https://github.com/lib/pq/pull/621
- https://github.com/lib/pq/pull/788
- https://github.com/lib/pq/pull/833

**Comment by jesperpedersen** [comment]
https://github.com/lib/pq/releases/tag/v1.1.1 contains the support for this

**Comment by RobAtticus** [comment]
Wanted to resurface this @wrouesnel because we have found it to be quite important in a project we're working on. I see there's a PR (#278) to address this, although it's biting off a lot in addition. Is it possible to get this change in by itself, which seems to be just updating the pq dependency to 1.1.1?

**Comment by jesperpedersen** [comment]
Yeah, and https://github.com/lib/pq/releases/tag/v1.2.0 is out now

**Comment by wrouesnel** [comment]
Resolved in v0.6.0 and latest.

**Comment by brunolnetto** [comment]
Hi, folks. I start postgres_exporter.service on Linux 20.04. I am unable to connect to the target database. Hhe resuling log is `pq: unknown authentication response: 10`. It seems, there is some issue with `SCRAM-SHA-256` I am not aware of. May you provide the steps or commands necessary to resolvethis issue?
